packages/create-redwood-app/template/api/src/lib/prisma.ts

```diff
@@ -0,0 +1,91 @@
+import { PrismaClient, Prisma } from '@prisma/client'
+
+import { logger } from './logger'
+
+// To help you identify particularly slow parameter values,
+// For example, Heroku outputs the slowest queries (that take 2 seconds or more)
+const SLOW_QUERY_THRESHOLD = 2_000 // 2 seconds
+
+/*
+ * Determines the type and level of logging.
+ *
+ * See: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#log
+ */
+export const defaultLogLevels: Prisma.LogLevel[] = ['info', 'warn', 'error']
+
+/*
+ * Generates the Prisma Log Definitions from log levels
+ *
+ * @return Prisma.LogDefinition[]
+ */
+export const prismaLoggerOptions = (
+  setLogLevels: Prisma.LogLevel[]
+): Prisma.LogDefinition[] => {
+  return setLogLevels?.map((level) => {
+    return { emit: 'event', level } as Prisma.LogDefinition
+  })
+}
+
+/*
+ * To use event-based logging:
+ * Set emit to event for a specific log level, such as query
+ * Use the $on() method to subscribe to the event
+ *
+ * https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging
+ */
+export const configureLogLevels = (
+  client: PrismaClient,
+  setLogLevels?: Prisma.LogLevel[]
+) => {
+  setLogLevels?.forEach((level) => {
+    if (level === 'query') {
+      client.$on(level, (queryEvent: Prisma.QueryEvent) => {
+        if (queryEvent.duration >= SLOW_QUERY_THRESHOLD) {
+          logger.warn(
+            { ...queryEvent },
+            `Slow Query performed in ${queryEvent.duration} msec`
+          )
+        } else {
+          logger.debug(
+            { ...queryEvent },
+            `Query performed in ${queryEvent.duration} msec`
+          )
+        }
+      })
+    } else {
+      client.$on(level, (logEvent: Prisma.LogEvent) => {
+        switch (level) {
+          case 'info':
+            logger.info({ ...logEvent }, logEvent.message)
+            break
+          case 'warn':
+            logger.warn({ ...logEvent }, logEvent.message)
+            break
+          case 'error':
+            logger.error({ ...logEvent }, logEvent.message)
+            break
+          default:
+            logger.info({ ...logEvent }, logEvent.message)
+        }
+      })
+    }
+  })
+
+  return
+}
+
+/*
+ * Generates the Prisma Log Definitions from log levels
+ *
+ * @return PrismaClient
+ */
+export const createPrismaClient = (setLogLevels?: Prisma.LogLevel[]) => {
+  const client = new PrismaClient({
+    log: prismaLoggerOptions(setLogLevels),
+    errorFormat: 'colorless',
+  })
+
+  setLogLevels && configureLogLevels(client, setLogLevels)
+
+  return client
+}
```

Comment:
@dthyresson I know this has now merged, but I was wondering if we could hide most of this implementation?

The reason is I'm worried that Prisma changes the logging interfaces and we have to ship a huge code mod. 

Maybe there's a way to internalize `configureLogLevels`?

Reply:
@peterp 

Definitely would like to.

I commented yesterday above:

> > Can we hide these implementations in the logger library in the api package?
> 
> @peterp I didn't know if I wanted to have the api/logger depend on Prisma.
> 
> Some deploys won't have Prisma so then would that prevent people from using the logger?
> 
> > I'm concerned that Prisma changes the format
> 
...
> 
> But not sure I can think of an alternative.

But of course when you say:

> The reason is I'm worried that Prisma changes the logging interfaces and we have to ship a huge code mod

I definitely want to avoid that scenario.

I am having a hard time visualizing what

>  if we could hide most of this implementation? 

or

> internalizing 

it might look like?

Can you give me a direction or maybe we jump of a quick call?

I wanted to get logging in so people could give feedback like this and can improve the implementation so can assess if we want to ship the CRWA template changes in w/ v0.28.